render.go

```diff
@@ -195,13 +196,19 @@ func (r *Render) Render(buffer *Buffer, previousText string, lastKeyStroke Key,
 	// prepare area by getting the end position the console cursor will be at after rendering
 	cursorEndPos := r.getCursorEndPos(prefix+line, 0)
 
+	// If the user writes something, we clear diagnostics (highlights and error shown) because the ranges might be outdated
+	if buffer.Text() != previousText {
+		r.diagnostics = nil
+	}
+
```

Comment:
If we clear the diagnostics anyway, I don't see the point of keeping it in Render as a field at all. Couldn't we just pass it to the render function directly? So basically this:

```
func (p *Prompt) SetDiagnostics(diagnostics []lsp.Diagnostic) {
	p.renderer.Render(p.buf, p.buf.Text(), p.lastKey, p.completion, p.lexer, diagnostics)
}
```

Maybe I'm misunderstanding something though

Reply:
We only clear it when the text is changed, but not when the user navigates it with arrows, for example. That's the case where we need it stored so we can continue rendering it. 

The UX feels much better when you can navigate the statement and the error doesn't disappear immediately, and you forget where it was. That's the reason why I implemented it like this

Reply:
Ah yeah okay keeping it in memory makes sense then! 

Could we move it to the prompt object though? Because I'm not sure it should belong to the Render struct responsibility-wise. In the same way the lexer is part of prompt too

Reply:
I've refactored this. Still wondering if the result is much better. Let me know if you think